### ISSUE TITLE
Full UI navigation via keyboard only

### ISSUE THREAD

**Issue by quincylvania** [openstreetmap/iD#8004]
[Keyboard navigation](https://webaim.org/techniques/keyboard/) is an important factor in accessibility. Blind people and those with motor control difficulties often rely on keyboards or equivalent devices to use the web. iD has not yet been optimized for these experiences.

While #7770 removed keytraps, more work is needed to make keyboards first-class input devices. To close this issue, users should be able to navigate to all applicable UI elements in iD using only a keyboard. This doesn't mean they'll necessarily be able to execute all of iD's functions, like drawing geometry, but that will be future work.

**Comment by quincylvania** [comment]
Putting aside the map, which has no focus support, I was able to significantly improve keyboard navigation in the rest of the UI. Previously, there were lots of unfocusable controls, backwards ordering, non-trapping modals, and controls with no focus styling.

Since we've gone from non-usable to usable, I think we can close this issue. But there's still more to be done to make keyboards first-class input devices, primarily enabling map navigation and editing. There's also the issue of screen reader support—all controls need accessibility labels and descriptions. These can be future work.
